chainstate/src/rpc/types/token.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{
+    address::{AddressError, RpcAddress},
+    chain::{
+        tokens::{
+            IsTokenFreezable, IsTokenUnfreezable, NftIssuance, TokenIssuance, TokenTotalSupply,
+        },
+        ChainConfig, Destination,
+    },
+    primitives::amount::RpcAmountOut,
+};
+
+#[derive(Debug, Clone, serde::Serialize)]
+pub enum RpcTokenTotalSupply {
```

Comment:
This file needs to be improved, significantly so. You should be using RpcString, not String, for strings that are expected to be ASCII. Also there's enums there without keys/values. Please take another go at it.

I pushed a few fixes in other places. We're getting close.